lib/pixelutil.ex

```diff
@@ -3,51 +3,57 @@ defmodule Inky.PixelUtil do
   `PixelUtil` maps pixels to bitstrings to be sent to an Inky screen
   """
 
-  def pixels_to_bitstring(pixels, display, color) do
-    rotation = display.rotation / 90
-    width = display.width
-    height = display.height
-
-    p2b =
-      case color do
-        :black -> &pixel_to_black/1
-        _ -> &pixel_to_accent/1
-      end
-
-    opts = bitstring_traversal_opts(rotation, width, height)
-    do_pixels_to_bitstring(pixels, p2b, opts)
+  def pixels_to_bits(pixels, width, height, rotation_degrees, color_map) do
+    opts = bitstring_traversal_opts(rotation_degrees, width, height)
+    do_pixels_to_bits(pixels, color_map, opts)
   end
 
-  defp pixel_to_black(:black), do: 0
-  defp pixel_to_black(_), do: 1
+  @doc """
+  Only exposed for testing purposes. Do not use.
 
-  defp pixel_to_accent(c) when c in [:red, :yellow, :accent], do: 1
-  defp pixel_to_accent(_), do: 0
+  ## Doctests
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(0, 212, 104)
+      {:y_outer, 0, 103, 0, 211}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(180, 212, 104)
+      {:y_outer, 103, 0, 211, 0}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(-90, 212, 104)
+      {:x_outer, 211, 0, 0, 103}
+
+      iex> Inky.PixelUtil.bitstring_traversal_opts(90, 212, 104)
+      {:x_outer, 0, 211, 103, 0}
+
+  """
+  def bitstring_traversal_opts(rotation_degrees, width, height) do
+    # Simplify and wrap around rotations
+    rotation =
+      rotation_degrees
+      |> div(90)
+      |> rem(4)
+      |> (&if(&1 < 0, do: &1 + 4, else: &1)).()
```

Comment:
Math 😎

Reply:
I followed until the rem(4) which I'm uncertain of what it does and then we get an if that is perl to me :) Can we clarify the why's a bit?
There is some ASCII-art towards the end too. Calling something, certainly :)
I'm sure it is a great improvement but how would I know?

Reply:
Ok, we've gone through it. If will get clarified slightly to be more syntactically readable and the rest is fine.

Reply:
I'll change it to something like `(fn r -> if(r < 0, do: r + 4, else: r) end).()`